Core/TextSizeUserScript.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  TextSizeUserScript.swift
+//  DuckDuckGo
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+import WebKit
+
+public class TextSizeUserScript: NSObject, UserScript {
+
+    public var textSizeAdjustmentInPercents: Int = 100
+    
+    public weak var delegate: PrintingUserScriptDelegate?
+
+    public var source: String {
+        let dynamicTypeScalePercentage = UIFontMetrics.default.scaledValue(for: 100)
+        
+        return """
+        (function() {
+            let topLevelUrl = getTopLevelURL();
+            let shouldAdjustForDynamicType = topLevelUrl.hostname.endsWith("wikipedia.org");
```

Comment:
Do we ever expect to have more domains to need this? Would be nice to manage this more as a dynamic list. Similar to how exceptions are injecting into the fingerprinting script.